### PR TITLE
Libraries: Change enums to enum classes and constexpr variables

### PR DESCRIPTION
**Pull request by ldm5180**
- LibChess
- LibCompress

### CODE REVIEW

**Review by kleinesfilmroellchen — APPROVED** [review]
🧹

File: Userland/Libraries/LibCompress/Deflate.h [hunk 1]
```diff
@@ -103,7 +103,7 @@ class DeflateDecompressor final : public InputStream {
     CircularDuplexStream<32 * KiB> m_output_stream;
 };
 
-enum DeflateSpecialCodeLengths : u32 {
+enum class DeflateSpecialCodeLengths : u32 {
```
Comment (IdanHo) [inline Userland/Libraries/LibCompress/Deflate.h:106]: Not sure I love this, the majority of users of the enum deal with the raw versions, meaning it's all static casts and to_underlying()s

**Comment by ldm5180** [comment]
Good point. Makes me think this shouldn’t be an enum at all. Maybe just 3
constexpr values.

File: Userland/Libraries/LibCompress/Deflate.cpp [hunk 2]
```diff
@@ -16,6 +16,10 @@
 
 namespace Compress {
 
+static constexpr u8 deflate_special_code_length_copy = 16;
```
Comment (IdanHo) [inline Userland/Libraries/LibCompress/Deflate.cpp:19]: This does look better, but please update the commit title accordingly :^)

**Review by linusg — COMMENTED** [review]
Sorry to commit-message-nitpick again, but you can be much more specific when the change doesn't span across multiple *things* or even libraries. Something like

> LibChess: Change UCI::Command::Type to an enum class

**Comment by ldm5180** [comment]
@linusg Done.

> Sorry to commit-message-nitpick again, but you can be much more specific when the change doesn't span across multiple _things_ or even libraries. Something like
> 
> > LibChess: Change UCI::Command::Type to an enum class

**Pull request merged**